compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java

```diff
@@ -139,6 +166,16 @@ private JBallerinaBackend(PackageCompilation packageCompilation, JvmTarget jdkVe
         this.interopValidator = InteropValidator.getInstance(compilerContext);
         this.jvmCodeGenerator = CodeGenerator.getInstance(compilerContext);
         this.conflictedJars = new ArrayList<>();
+        this.symbolTable = SymbolTable.getInstance(compilerContext);
+        this.usedBIRNodeAnalyzer = UsedBIRNodeAnalyzer.getInstance(compilerContext);
+        this.codeGenOptimizationReportEmitter = CodeGenOptimizationReportEmitter.getInstance(compilerContext);
+        if (packageCompilation.compilationOptions().optimizeCodegen()) {
```

Comment:
In the else case shall we set all of these as `Collections.emptyMap`? That way when people access these fields they don't need think about the initialization state (or check if the field is nil). Given we are anyway going to spend memory to store the null pointer and emptyMap being a singleton I don't think we are saving much by setting them as null. Also make see if you can make these fields `final` afterward.

Reply:
Used `Collections.emptyMap` and `Collections.emptySet` to initialize them for else case. Added final keyword as well. Changes can be found in c8c56fff75d0077964eb8a793986bb68f3e8a915.